app/uk/gov/hmrc/hec/services/IFService.scala

```diff
@@ -119,40 +118,78 @@ class IFServiceImpl @Inject() (
           httpResponse
             .parseJSON[RawCTSuccessResponse]
             .leftMap(e => BackendError(Error(e)))
-            .flatMap(response =>
-              Either
-                .fromOption(
-                  CTStatus.fromString(response.returnStatus),
-                  BackendError(Error(s"Could not parse success return status ${response.returnStatus}"))
-                )
-                .map(CTStatusResponse(utr, startDate, endDate, _, response.accountingPeriods))
-            )
+            .flatMap(ctAccountingPeriodsValidation(_).map { accountingPeriods =>
+              CTStatusResponse(utr, startDate, endDate, latestAccountingPeriod(accountingPeriods))
+            })
         } else {
           handleErrorPath(httpResponse, utrType = "CT")
         }
       }
+
+  private def latestAccountingPeriod(accountingPeriods: List[AccountingPeriod]): Option[AccountingPeriod] =
+    accountingPeriods.sortWith { case (a1, a2) => a1.endDate.isAfter(a2.endDate) }.headOption
+
+  private def ctAccountingPeriodsValidation(
+    response: RawCTSuccessResponse
+  ): Either[BackendError, List[AccountingPeriod]]                                                         =
+    if (response.returnStatus === "0") // 0 = successful
+      response.accountingPeriods
+        .getOrElse(List.empty[RawAccountingPeriod])
+        .traverse[Either[BackendError, *], AccountingPeriod](a =>
+          toCtStatus(a)
+            .map(status => AccountingPeriod(a.accountingPeriodStartDate, a.accountingPeriodEndDate, status))
+        )
+        .filterOrElse(
+          _.nonEmpty,
+          BackendError(Error("Could not find accounting periods for outer return status '0 = successful'"))
+        )
+    else if (response.returnStatus === "2") // 2 = No live records exist for the input parameters provided
+      Right(List.empty)
+    else
+      Left(BackendError(Error(s"Could not parse outer return status ${response.returnStatus}")))
+
+  private def toCtStatus(rawAccountingPeriod: RawAccountingPeriod): Either[BackendError, CTStatus] =
+    rawAccountingPeriod.accountingPeriodStatus match {
+      case "1"   => Right(CTStatus.ReturnFound)
+      case "2"   => Right(CTStatus.NoticeToFileIssued)
+      case "3"   => Right(CTStatus.NoReturnFound)
+      case other => Left(BackendError(Error(s"Could not parse accounting period status $other")))
```

Comment:
does the api return a string for status? just a bit weird that if they have chosen to return an integer, then why put it in a string

Reply:
they've got both a text description field and status number for the inner accounting periods, but for the outer `returnStatus` there's only a number from what I can see. There's less room for error on a number so I thought I'd just use the number in the inner status in the accounting periods